osmdroid-android/src/main/java/org/osmdroid/util/RectL.java

```diff
@@ -316,4 +316,40 @@ public static long getRotatedY(final long pX, final long pY,
                                    final double pCos, final double pSin) {
         return pCenterY + Math.round((pX - pCenterX) * pSin + (pY - pCenterY) * pCos);
     }
+
+    /**
+     * @since 6.0.2
+     */
+    public void offset(final long pDx, final long pDy) {
+        left += pDx;
+        top += pDy;
+        right += pDx;
+        bottom += pDy;
+    }
+
+    /**
+     * @since 6.0.2
+     */
+    public void union(final long pLeft, final long pTop, final long pRight, final long pBottom) {
+        if ((pLeft < pRight) && (pTop < pBottom)) {
+            if ((left < right) && (top < bottom)) {
```

Comment:
Oh, I just copied the code from `android.graphics.Rect` with `long` instead of `int`.
> Shouldn't a check like this `if ((left < right) && (top < bottom))` be done beforehand, when RectL is created?

There's no such thing in the original `Rect` code. When I created the `RectL` class I had to deal with all the osmdroid code history with `Rect`, and it was too dangerous to change the behavior. And I'm not against a `left > right` possibility in order to handle the date line cases, but it's a matter of taste.

Reply:
If a Rect with `left >= right`or `top >= bottom` are valid usage cases, then OK.